x-pack/plugins/canvas/public/components/app/index.js

```diff
@@ -36,28 +45,24 @@ const mapStateToProps = state => {
   };
 };
 
-const types = {
-  elements: elementsRegistry,
-  transformUIs: transformRegistry,
-  datasourceUIs: datasourceRegistry,
-  modelUIs: modelRegistry,
-  viewUIs: viewRegistry,
-  argumentUIs: argTypeRegistry,
-  templates: templatesRegistry,
-  tagUIs: tagsRegistry,
-};
+elementSpecs.forEach(spec => elementsRegistry.register(spec));
```

Comment:
All this work shouldn't be happening in the component file, but in `public/app.js`